src/main/java/com/google/firebase/remoteconfig/Version.java

```diff
@@ -37,37 +40,53 @@
  */
 public final class Version {
 
-  private String versionNumber;
-  private long updateTime;
-  private String updateOrigin;
-  private String updateType;
-  private User updateUser;
-  private String description;
-  private String rollbackSource;
-  private boolean legacy;
+  private static final Pattern ZULU_TIME_NO_NANOSECONDS_PATTERN = Pattern
+          .compile("^(\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2})");
 
-  /**
-   * Creates a new {@link Version} with a description.
-   */
-  public static Version withDescription(String description) {
-    return new Version().setDescription(description);
-  }
+  private final String versionNumber;
+  private final long updateTime;
+  private final String updateOrigin;
+  private final String updateType;
+  private final User updateUser;
+  private final String rollbackSource;
+  private final boolean legacy;
+  private String description;
 
-  Version() {
+  private Version() {
+    this.versionNumber = null;
+    this.updateTime = 0L;
+    this.updateOrigin = null;
+    this.updateType = null;
+    this.updateUser = null;
+    this.rollbackSource = null;
+    this.legacy = false;
   }
 
   Version(@NonNull VersionResponse versionResponse) {
     checkNotNull(versionResponse);
     this.versionNumber = versionResponse.getVersionNumber();
+
     if (!Strings.isNullOrEmpty(versionResponse.getUpdateTime())) {
-      SimpleDateFormat dateFormat = new SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss z");
+      // Update Time is a timestamp in RFC3339 UTC "Zulu" format, accurate to nanoseconds.
+      // example: "2014-10-02T15:01:23.045123456Z"
+      // SimpleDateFormat cannot handle nanoseconds, therefore we drop nanoseconds from the string.
+      Matcher errorMatcher = ZULU_TIME_NO_NANOSECONDS_PATTERN
```

Comment:
Instead of a complex regex, why not just strip away everything (and including) the period? That seems easier and more readable.

Reply:
Makes sense! Updated the code to use `substring()`